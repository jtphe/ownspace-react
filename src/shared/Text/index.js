import React from 'react';

import { Text, StyleSheet } from 'react-native';
import { useFonts } from '@use-expo/font';

const TextHelvetica = props => {
  useFonts({
    // eslint-disable-next-line global-require
    HelveticaNeue: require('../../../assets/fonts/HelveticaNeue.ttf')
  });

  const { style, ellipsizeMode, selectable, children } = props;
  return (
    <Text
      style={[style, styles.fontStyle]}
      ellipsizeMode={ellipsizeMode}
      selectable={selectable}
    >
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  fontStyle: { fontFamily: 'HelveticaNeue' }
});

export default TextHelvetica;
