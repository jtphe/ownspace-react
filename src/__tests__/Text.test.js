import 'react-native';
import React from 'react';
import Text from '@shared/Text';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Text tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Text />);
  });

  describe('Autocomplete snapshot', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    renderer.create(<Text />);
  });
});

Enzyme.configure({ adapter: new Adapter() });
