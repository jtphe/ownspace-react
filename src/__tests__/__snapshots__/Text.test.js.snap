// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Text
  style={
    Array [
      undefined,
      undefined,
    ]
  }
/>
`;
